### PR TITLE
BEG-133 - Create module to handle authorisation of introspection queries

### PR DESCRIPTION
**Pull request by aligent-lturner**
Adds the ability to prevent introspection queries by users without appropriate authorisation. This allows for introspection to remain enabled for things such as FE build processes, but without it being allowed for general users.

### CODE REVIEW

**Comment by indunil15** [comment]
LGTM

**Review by indunil15 — APPROVED** [review]
LGMT

**Review by dasithaaligent — APPROVED** [review]
LGTM!

File: Model/Config.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    /**
+     * Indicates if introspection authorisation is enabled or disabled
+     *
+     * @return bool
+     */
+    public function getIntrospectionAuthEnabled(): bool
+    {
+        return $this->scopeConfig->isSetFlag(self::XML_PATH_INTROSPECTION_AUTH_ENABLED);
```
Comment (brettlaishley) [inline Model/Config.php:31]: Won't this return `null` until the value is saved once, and potentially throw a type error?

**Review by brettlaishley — COMMENTED** [review]
LGTM, <del>just one question though</del>.

File: Model/Config.php (hunk 1 above)
Comment (brettlaishley) [reply Model/Config.php:31]: You can ignore this, I misread `isSetFlag` for the usual `getValue` function from scope config.

**Review by brettlaishley — APPROVED** [review]

**Pull request merged**